### PR TITLE
Porting to vegman

### PR DESCRIPTION
**Pull request by nest1ing**
This patch set fixes `yadro-snmp` build for Vegman and makes SNMP tables work properly.
But a couple of issues still present on Vegman, that will be solved with the further path sets.

### CODE REVIEW

File: agent/yadro/sensors.cpp [hunk 1]
```diff
@@ -93,15 +91,14 @@ struct Sensor
...
-        phosphor::snmp::data::table::Item<int64_t, int64_t, bool, int64_t, bool,
-                                          int64_t, bool, int64_t, bool,
-                                          int64_t>(folder, name,
-                                                   0,        // Value
-                                                   0, false, // WarningLow
-                                                   0, false, // WarningHigh
-                                                   0, false, // CriticalLow
-                                                   0, false, // CriticalHigh
-                                                   1)        // Scale
+        phosphor::snmp::data::table::Item<double, double, bool, double, bool,
```
Comment (alatarum) [inline agent/yadro/sensors.cpp:94]: Don't we need to support old platforms? How does it work now?

File: configure.ac [hunk 2]
```diff
@@ -23,6 +23,11 @@ PKG_CHECK_MODULES([SDBUSPLUS], [sdbusplus],,\
 AC_SUBST(SDBUSPLUS_CFLAGS)
 AC_SUBST(SDBUSPLUS_LIBS)
 
+PKG_CHECK_MODULES([SDEVENTPLUS], [sdeventplus],,\
+    AC_MSG_ERROR(["Required sdeventplus package."]))
```
Comment (artemsen) [inline configure.ac]: Antipattern: https://github.com/openbmc/docs/blob/master/anti-patterns.md#explicit-ac_msg_error-on-pkg_check_modules-failure

File: agent/main.cpp [hunk 3]
```diff
@@ -101,45 +103,49 @@ int parse_args(int argc, char** argv)
...
+    sdbusplus::helper::helper::getBus().attach_event(evt.get(),
+                                                     SD_EVENT_PRIORITY_NORMAL);
 
-    TRACE_INFO("%s is up and running.\n", PACKAGE_STRING);
-    snmpagent_init();
+    sigset_t ss;
+    if (sigemptyset(&ss) < 0 || sigaddset(&ss, SIGTERM) < 0 ||
+        sigaddset(&ss, SIGINT) < 0)
+    {
+        TRACE_ERROR("Failed to setup signals hanlders.\n");
```
Comment (AlexanderAmelkin) [inline agent/main.cpp]: signals -> signal

File: agent/main.cpp [hunk 4]
```diff
@@ -101,45 +103,49 @@ int parse_args(int argc, char** argv)
     return rc;
 }
 
-void setup_signals(sdevent::event::Event& evt)
```
Comment (AlexanderAmelkin) [inline agent/main.cpp:104]: Why did you decide to move it from a helper function into main() ?

File: agent/yadro/sensors.cpp (hunk 1 above)
Comment (AlexanderAmelkin) [reply agent/yadro/sensors.cpp:94]: I guess old platforms use old version of this repo. And if we decide to upgrade them, we'll upgrade the interface as well, just as we do when merging updates from upstream for other projects.

File: agent/yadro/sensors.cpp [hunk 5]
```diff
@@ -131,14 +128,11 @@ struct Sensor
             }
 
             // Correct scale power
+            // Required for TEXTUAL-CONVENTION in YADRO-MIB.txt
             switch (folder[n + 1])
             {
-                case ST_TEMPERATURE:
-                case ST_VOLTAGE:
-                case ST_CURRENT:
-                case ST_POWER:
```
Comment (AlexanderAmelkin) [inline agent/yadro/sensors.cpp:139]: What happened to these values? Why are they not needed anymore?

File: agent/yadro/sensors.cpp (hunk 1 above)
Comment (nest1ing) [reply agent/yadro/sensors.cpp:94]: Yes, exactly so.

File: configure.ac (hunk 2 above)
Comment (nest1ing) [reply configure.ac]: fixed

File: agent/main.cpp (hunk 4 above)
Comment (nest1ing) [reply agent/main.cpp:104]: There are two objects (for each signal) that should be alive during main loop working.
And this function is part of their initialization and called only once.

**Review by artemsen — DISMISSED** [review]

File: agent/yadro/sensors.cpp (hunk 5 above)
Comment (nest1ing) [reply agent/yadro/sensors.cpp:139]: The logic was changed. I would say that the logic was inverted. Now only `ST_TACHOMETER` requires the correction.

File: agent/main.cpp (hunk 3 above)
Comment (nest1ing) [reply agent/main.cpp]: fixed

**Review by AlexanderAmelkin — APPROVED** [review]

**Review by alatarum — APPROVED** [review]

**Pull request merged**